File: libraries/hlp/length.f
      INTEGER FUNCTION hlp_LENGTH (STRING)
*+
*  - - - - - - -
*   L E N G T H
*  - - - - - - -
*
*  Length of a string excluding any trailing spaces.
*
*  Given:
*     STRING       c*(*)    the string
*
*  Returned:
*     hlp_LENGTH   i        length excluding trailing spaces
*
*  The minimum length returned is 1.
*
*  P.T.Wallace   Starlink   24 February 1991
*-

      IMPLICIT NONE

      CHARACTER STRING*(*)

      INTEGER L,I



      L=LEN(STRING)
      I=1
      DO WHILE (I.LE.L.AND.STRING(I:).NE.' ')
         I=I+1
      END DO
      hlp_LENGTH=MAX(1,I-1)

      END

File: libraries/hlp/length.c
#include <ctype.h>
int hlpLength ( char *string )
/*
**  - - - - - - - - - -
**   h l p L e n g t h
**  - - - - - - - - - -
**
**  Length of a string excluding any trailing whitespace.
**
**  Given:
**     *string      char     the string
**
**  Returned (function value):
**                  int      length excluding trailing spaces
**
**  The minimum length returned is 0.
**
**  Last revision:   7 January 1996
**
**  Copyright 1996 P.T.Wallace.  All rights reserved.
*/
{
   int l, i, c;

   l = 0;
   for ( i = 0;
         ( c = (int) string [ i++ ] );
         l = isspace ( c ) ? l : i );
   return l;
}
